src/handlers/ps2census/DeathEventHandler.ts

```diff
@@ -0,0 +1,75 @@
+import {injectable, multiInject} from 'inversify';
+import {getLogger} from '../../logger';
+import {jsonLogOutput} from '../../utils/json';
+import DeathEvent from './events/DeathEvent';
+import {TYPES} from '../../constants/types';
+import CharacterPresenceHandler from '../CharacterPresenceHandler';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {Death} from 'ps2census';
+import Parser from '../../utils/parser';
+import PS2EventQueueMessage from '../messages/PS2EventQueueMessage';
+import {PS2EventInstanceHandlerContract} from '../../interfaces/PS2EventInstanceHandlerContract';
+import AggregateHandlerInterface from '../../interfaces/AggregateHandlerInterface';
+import CharacterBroker from '../../brokers/CharacterBroker';
+import ItemBroker from '../../brokers/ItemBroker';
+
+@injectable()
+export default class DeathEventHandler implements PS2EventInstanceHandlerContract<Death> {
+    public readonly eventName = 'Death';
+    private static readonly logger = getLogger('DeathEventHandler');
+
+    constructor(
+        private readonly itemBroker: ItemBroker,
+        private readonly characterBroker: CharacterBroker,
+        private readonly characterPresenceHandler: CharacterPresenceHandler,
+        @multiInject(TYPES.deathAggregates) private readonly aggregateHandlers: Array<AggregateHandlerInterface<DeathEvent>>,
+    ) {}
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async handle(event: PS2EventQueueMessage<Death>): Promise<boolean> {
```

Comment:
> and the acknowledgement can be done here.

Yes that's indeed true, we could pack the channel actions to the level below. Wouldn't it be more of code duplication though because I can't expect the logic would change between each message type? It either processes successfully and returns true, if it returns false it needs dropping and if an exception is thrown requeue, and that would be the same across all message types no?

Reply:
Code duplication will be at a minimum and you retain some flexibility for later, up to you.

Reply:
For now I'm keeping it KISS, I shouldn't need to handle acks lower down the chain as of yet, may need to do later on though